k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -193,129 +194,36 @@ type DeviceClaimStatus struct {
+	// is owned by the driver.
+	//
 	// +optional
-	Allocations []DevicePoolAllocation `json:"allocations,omitempty"`
-
-	// PodNames contains the names of all Pods using this claim.
-	// TODO: Can we just use ownerRefs instead?
-	// +optional
-	PodNames []string `json:"podNames,omitempty"`
-}
-
-// DevicePrivilegedClaim is used to specify a special kind of privileged claim
-// for a set of devices on a node. This type of claim is used for monitoring or
-// other management services for a device. It ignores all ordinary claims to
-// the device with respect to access modes and any resource allocations. As a
-// separate type, it can (and is expected to) have separate RBAC constraints.
-//
-// It does not have all the sophisticated selection mechanisms of an ordinary
-// device claim, as the most common use case is simply to access all devices
-// managed by a given driver on a given node. It intentionally does not require
-// a class, though it does allow some flexibility with the specification of
-// Constraints and Configs.
-
-type DevicePrivilegedClaim struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   DevicePrivilegedClaimSpec   `json:"spec,omitempty"`
-	Status DevicePrivilegedClaimStatus `json:"status,omitempty"`
-}
-
-// DevicePrivilegedClaimSpec contains the details of the privileged claim.
-type DevicePrivilegedClaimSpec struct {
-	// Driver will limit the scope of devices considered to only those
-	// published by the specified driver.
-	// +required
-	Driver string `json:"driver,omitempty"`
-
-	// Constraints is a CEL expression that operates on device attributes.
-	// Only devices matching this constraint will be selected by this
-	// claim.
-	// +optional
-	Constraints *string `json:"constraints,omitempty"`
-
-	// Configs contains references to arbitrary vendor device configuration
-	// objects that will be attached to the device allocation.
-	// +optional
-	Configs []DeviceConfigReference `json:"configs,omitempty"`
-}
-
-// DevicePrivilegedClaimStatus contains the results of the claim allocation.
-type DevicePrivilegedClaimStatus struct {
-	// ClaimConfigs contains the entire set of dereferenced vendor
-	// configurations from the DeviceClaim, as of the time of allocation.
-	// +optional
-	ClaimConfigs []runtime.RawExtension `json:"claimConfigs,omitempty"`
+	Allocations []DeviceAllocation `json:"allocations,omitempty"`
 
-	// Allocations contains the list of device allocations needed to
-	// satisfy the claim, one per pool from which devices were allocated.
-	// +optional
-	Allocations []DevicePoolAllocation `json:"allocations,omitempty"`
+	// Devices contains the status of each device assigned to this
+	// claim, as reported by the driver. This can include driver-specific
+	// information. Entries are owned by their respective drivers.
+	// TODO: How can we do that?
+	DeviceStatuses []DeviceStatus `json:"deviceStatuses,omitempty"`
```

Comment:
Anyone feel that some drivers will wipe out other devices status here 😄 and start hot looping?